evennia/typeclasses/attributes.py

```diff
@@ -202,6 +202,21 @@ def __init__(self, default=None, category=None, strattr=False, lockstring="", au
         self._lockstring = lockstring
         self._autocreate = autocreate
         self._key = ""
+        
+    @property
+    def _default(self):
+        """
+        Tries returning a new instance of default if callable.
+        
+        """
+        try:
+            return self.__default()
+        except:
+            return self.__default
```

Comment:
Never use a 'bare' `except` like this, it's frowned upon in PEP8 standard and is not considered good practice  (always use at least `except Exception`). 

But in this case I think it's also rather motivated to do a `callable` check here instead; it's probably more likely to not have a callable than having one, and relying on catching the exception every time is comparably expensive (even if 50% are callable, it's still about 10x faster to check `callable()` than to use `try...except` here). 

```python 
if callable(self.__default):
    return self.__default()
else:
    return self.__default
```